### PR TITLE
Add automatic cert generation for interoperability package

### PR DESCRIPTION
**Pull request by drizzentic**
## Summary by CodeRabbit

- **New Features**
	- Introduced a configuration file for Traefik, enhancing service management in a microservices architecture.
	- Enabled secure communication (TLS) for the Keycloak service and improved routing for the OpenHIM interoperability layer.
	- Added the Traefik dashboard for better monitoring and management of the reverse proxy.

- **Security Enhancements**
	- Implemented automated SSL certificate management.
	- Adopted HTTPS for the OpenHIM console, improving data security.
	- Added authentication requirements for access to the Traefik dashboard.

- **Performance Improvements**
	- Increased resource limits for the Traefik reverse proxy to handle higher loads effectively.

### CODE REVIEW

File: interoperability-layer-openhim/docker-compose.yml [hunk 1]
```diff
@@ -50,17 +50,22 @@ services:
...
         - traefik.http.services.openhimcore.loadbalancer.server.port=5000
         - traefik.http.services.openhimcore.loadbalancer.server.scheme=https
         - traefik.http.routers.openhimcore.tls=true
         - traefik.http.routers.openhimcore.entrypoints=websecure
         - traefik.http.routers.openhimcore.rule=Host(`${DOMAIN_NAME_HOST_TRAEFIK}`) && PathPrefix(`/openhimcore`)
-        - traefik.http.routers.openhimcore.middlewares=openhimcore
-        - traefik.http.middlewares.openhimcore.stripprefix.prefixes=/openhimcore
+        - traefik.http.middlewares.openhimcore-stripprefix.stripprefix.prefixes=/openhimcore
+        - traefik.http.routers.openhimcore.middlewares=openhimcore-stripprefix
+        - "traefik.http.routers.openhimcore.tls.certresolver=le"
```
Comment (brett-onions) [inline interoperability-layer-openhim/docker-compose.yml:65]: Its this config meant to be in quotations?

File: interoperability-layer-openhim/docker-compose.yml [hunk 2]
```diff
@@ -90,7 +95,8 @@ services:
         - traefik.http.services.openhim-console.loadbalancer.server.scheme=http
         - traefik.http.routers.openhim-console.service=openhim-console
         - traefik.http.routers.openhim-console.entrypoints=websecure
-        - traefik.http.routers.openhim-console.rule=Host(`${OPENHIM_SUBDOMAIN}.${DOMAIN_NAME_HOST_TRAEFIK}`)
+        - 'traefik.http.routers.openhim-console.tls=true'
```
Comment (brett-onions) [inline interoperability-layer-openhim/docker-compose.yml:98]: Is this config meant to be in quotations?

File: reverse-proxy-traefik/package-metadata.json [hunk 3]
```diff
@@ -13,8 +13,13 @@
     "TK_MEMORY_LIMIT": "3G",
     "TK_MEMORY_RESERVE": "500M",
     "INSECURE_SKIP_VERIFY": "true",
-    "ENABLE_TRAEFIK_DASHBOARD": "false",
-    "PLACEMENT_ROLE_CONSTRAINTS": "leader",
-    "ACME_EMAIL": ""
+    "ENABLE_TRAEFIK_DASHBOARD": "true",
+    "PLACEMENT_ROLE_CONSTRAINTS": "manager",
+    "ACME_EMAIL": "",
+    "USERNAME": "admin",
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply reverse-proxy-traefik/package-metadata.json:19]: this is configurable through environment variables

**Review by brett-onions — APPROVED** [review]
LGMT

**Review by bradsawadye — APPROVED** [review]

**Pull request merged**